python/deltalake/writer.py

```diff
@@ -153,7 +153,7 @@ def write_deltalake(
         else:
             # Non-existant local paths are only accepted as fully-qualified URIs
             table_uri = "file://" + str(Path(table_or_uri).absolute())
-        table = try_get_deltatable(table_or_uri)
+        table = try_get_deltatable(table_or_uri, storage_options)
```

Comment:
I tried this myself by also adding:

```python
def try_get_deltatable(table_uri: str, storage_options: Dict[str, str], version: Optional[int] = None) -> Optional[DeltaTable]:
    try:
        return DeltaTable(table_uri=table_uri, storage_options=storage_options, version=version)
    except PyDeltaTableError as err:
        # TODO: There has got to be a better way...
        if "Not a Delta table" in str(err):
            return None
        elif "cannot find" in str(err):
            return None
        elif "No such file or directory" in str(err):
            return None
        else:
            raise
```

This will return `None` when the table does not exist, instead of failing for creds.

Reply:
> This will return None when the table does not exist, instead of failing for creds, but potentially also taking the version into account later...

I'm not sure what you are saying here. It's supposed to return None when the table does not exist. Are you saying that if it just fails to authenticate to the Object store it might also return None?

Reply:
Sorry, to be clear.

I'm saying the logic seem correct and yes it should return None in this case.